fetcher/ubuntu.go

```diff
@@ -20,46 +21,63 @@ const (
 )
 
 // FetchUbuntuVulnList clones vuln-list and returns CVE JSONs
-func FetchUbuntuVulnList() (entries []models.UbuntuCVEJSON, err error) {
+func FetchUbuntuVulnList() (iter.Seq2[models.UbuntuCVEJSON, error], int, error) {
 	// Clone vuln-list repository
 	dir := filepath.Join(util.CacheDir(), "vuln-list")
 	updatedFiles, err := git.CloneOrPull(ubuntuRepoURL, dir, ubuntuDir)
 	if err != nil {
-		return nil, xerrors.Errorf("error in vulnsrc clone or pull: %w", err)
+		return nil, 0, xerrors.Errorf("error in vulnsrc clone or pull: %w", err)
 	}
 
 	// Only last_updated.json
 	if len(updatedFiles) <= 1 {
-		return nil, nil
+		return nil, 0, nil
 	}
 
 	rootDir := filepath.Join(dir, ubuntuDir)
 	targets, err := util.FilterTargets(ubuntuDir, updatedFiles)
 	if err != nil {
-		return nil, xerrors.Errorf("failed to filter target files: %w", err)
+		return nil, 0, xerrors.Errorf("failed to filter target files: %w", err)
 	} else if len(targets) == 0 {
 		log15.Debug("Ubuntu: no update file")
-		return nil, nil
+		return nil, 0, nil
 	}
 	log15.Debug(fmt.Sprintf("Ubuntu updated files: %d", len(targets)))
 
-	err = util.FileWalk(rootDir, targets, func(r io.Reader, _ string) error {
-		content, err := io.ReadAll(r)
-		if err != nil {
-			return err
-		}
+	count, err := countUbuntuCVEs(rootDir, targets)
+	if err != nil {
+		return nil, 0, xerrors.Errorf("failed to count Ubuntu CVEs: %w", err)
+	}
+
+	return func(yield func(models.UbuntuCVEJSON, error) bool) {
+
+		err = util.FileWalk(rootDir, targets, func(r io.Reader, _ string) error {
+			content, err := io.ReadAll(r)
+			if err != nil {
+				return err
+			}
 
-		cve := models.UbuntuCVEJSON{}
-		if err = json.Unmarshal(content, &cve); err != nil {
-			return xerrors.Errorf("failed to decode Ubuntu JSON: %w", err)
+			cve := models.UbuntuCVEJSON{}
+			if err = json.Unmarshal(content, &cve); err != nil {
+				return xerrors.Errorf("failed to decode Ubuntu JSON: %w", err)
+			}
```

Comment:
```diff
:100644 100644 50af323 0000000 M	fetcher/ubuntu.go

diff --git a/fetcher/ubuntu.go b/fetcher/ubuntu.go
index 50af323..7c006ec 100644
--- a/fetcher/ubuntu.go
+++ b/fetcher/ubuntu.go
@@ -50,15 +50,9 @@ func FetchUbuntuVulnList() (iter.Seq2[models.UbuntuCVEJSON, error], int, error)
 	}
 
 	return func(yield func(models.UbuntuCVEJSON, error) bool) {
-
 		err = util.FileWalk(rootDir, targets, func(r io.Reader, _ string) error {
-			content, err := io.ReadAll(r)
-			if err != nil {
-				return err
-			}
-
 			cve := models.UbuntuCVEJSON{}
-			if err = json.Unmarshal(content, &cve); err != nil {
+			if err = json.NewDecoder(r).Decode(&cve); err != nil {
 				return xerrors.Errorf("failed to decode Ubuntu JSON: %w", err)
 			}
 

```

Reply:
Fixed in https://github.com/vulsio/gost/pull/291/commits/d8f7399608f44d27d383d84fcf96a50e4df742bb